workflow/installertest.groovy

```diff
@@ -0,0 +1,191 @@
+    return (imageName+append).replaceAll(':','-')
+}
+
+/**
+ * Extracts the components from an {@code 'artifact://full/path/to/job/buildNr#artifact.ext'} type url.
+ * @param url the url
+ */
+@NonCPS
+def getComponentsFromArtifactUrl(String url) {
+    def pattern = /^artifact:\/\/([\/\w-_\. ]+)\/(\d+)\/{0,1}#([\/\w-_\.]+)$/
+    def matcher = url =~ pattern
+    if (matcher) {
+        return [
+            item: matcher.group(1),
+            run: matcher.group(2),
+            artifact: matcher.group(3)
+        ]
+    } else {
+        throw new MalformedURLException("Expected format: 'artifact://full/path/to/job/buildNr#dir/artifact.ext' but got '${url}'")
+    }
+}
+
+/**
+ * Downloads an arifact to the workspace for further use in the flow.
+ * This is a stripped down version of a CloudBees workflow utility function
+ * It has been generified to use with artifacts besides jenkins.war (for packaging use)
+ * 
+ * Caveats:
+ * Needs to run inside a node block, and does not stash the artifact (avoids issues with big artifacts)
+ * For smaller artifacts, you may wish to stash them
+ * @param url the url to download
+ *              {@code 'artifact://' will be downloaded via CopyArtifact build step},
+ *              Example: {@code 'artifact://full/path/to/job/buildNr#artifact.ext'}.
+ *              Anything else will be downloaded via wget
+ */
+def fetchArtifact(String url) {
+  if (url == null) {
+    fail 'required parameter url is missing'
+  } else if (url.startsWith("artifact://")) {
+    echo "Fetching ${url} as artifact."
+    def comp = this.getComponentsFromArtifactUrl(url)
+    step([$class: 'CopyArtifact', filter: comp.artifact, projectName: comp.item, flatten: true, selector: [$class: 'SpecificBuildSelector', buildNumber: comp.run]])
+  } else {
+      echo "Fetching ${url} as URL file."
+      sh "wget -q ${url}"
+  }
+}
+
+// Pull down the artifacts, must run in a node block
+def fetchInstallers(String debFileUrl, String rpmFileUrl, String suseFileUrl) {
+ sh 'rm -rf installers'
+ dir('installers') {
+   sh 'rm -rf deb rpm suse'
+   dir('deb') {
+      fetchArtifact(debFileUrl)
+   }
+   dir('rpm') {
+      fetchArtifact(rpmFileUrl)
+   }
+   dir('suse') {
+      fetchArtifact(suseFileUrl)
+   }
+  }
+}
+
+/** Runs a series of shell commands inside a docker image
+* The output of each command set is saved to a specific file and archived
+* Errors are propagated back up the chain. 
+* @param imageName docker image name to use (must support sudo, based off the sudoable images)
+* @param shellCommands List of (string) shell commands to run within the container
+* @param stepNames List of names for each shell command step, optional
+*                    (if not supplied, then the step # will be used)
+*/
+def runShellTest(String imageName, def shellCommands, def stepNames=null) {
+  withEnv(['HOME='+pwd()]) { // Works around issues not being able to find docker install
```

Comment:
if you need to set home then this is a sign that your slave is not correctly installed.
in upstart both `HOME` and `USER` need to be defined for the slave.